CMakeLists.txt

```diff
@@ -0,0 +1,1109 @@
+cmake_minimum_required(VERSION 3.15.1)
+
+project(wasi-libc
+  LANGUAGES ASM C)
+
+include(CMakeDependentOption)
+
+option(ENABLE_THREADS "enable threads" OFF)
+option(BUILD_DLMALLOC "build dlmalloc" YES)
+cmake_dependent_option(BUILD_LIBC_BOTTOM_HALF "build libc bottom half" YES
+  "BUILD_DLMALLOC" YES)
+cmake_dependent_option(BUILD_LIBC_TOP_HALF "build libc top half" YES
+  "BUILD_LIBC_BOTTOM_HALF" YES)
```

Comment:
I honestly do not know.  I replicated the options because they exist in the make system.  If they are no longer useful, I would really prefer to remove the options.  The fewer options there are, the better it is.

Reply:
I added those options fairly early on with the idea that people may want to replace the "top half" with a different implementation, but reuse the "bottom half" which is where most of the WASI-specific code is, but I haven't heard of anyone actually interested in doing this, so I'd be ok removing the option.

Reply:
Okay, that sounds good to me.  I think that fewer options make things easier to maintain, Ill make a change on top of this to remove the option.

Reply:
Removed `BUILD_DLMALLOC`, `BUILD_LIBC_BOTTOM_HALF` and `BUILD_LIBC_TOP_HALF`.